### PR TITLE
rm pylint and add ruff linter and formatter

### PR DESCRIPTION
**Pull request by jkhales**
Would suggest removing pylint in a follow up

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -20,7 +20,10 @@ clean-docs:
 	rm -rf docs/user_guide/jupyter_execute
 
 lint:
-	$(BIN)python -m pylint pylabrobot
+	$(BIN)python -m ruff check pylabrobot && $(BIN)python -m pylint pylabrobot
```
Comment (rickwierenga) [inline Makefile:23]: do we need both?

File: .github/workflows/format.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      - name: Update packages
+        run: sudo apt-get update
+      - name: Set up Python 3.9
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.9
+      - name: Install Dependencies
+        run: pip install -e '.[dev]'
+      - name: Run formatting check
+        run: make format
```
Comment (rickwierenga) [inline .github/workflows/format.yml:26]: why both this and linting?

File: Makefile (hunk 1 above)
Comment (jkhales) [reply Makefile:23]: i'd love to get rid of pylint, but it seems like you did a lot of work on it. 

are you opposed to me just rm'ing it?

File: .github/workflows/format.yml (hunk 2 above)
Comment (jkhales) [reply .github/workflows/format.yml:26]: i can put them in the same workflos if you like

File: Makefile (hunk 1 above)
Comment (jkhales) [reply Makefile:23]: some differences are that ruff seems less strict about line len. makes pylint fail

Comment (rickwierenga) [reply Makefile:23]: it's mostly google's template, and i added some small things. as long as behavior is roughly the same it should be good to rm here

Comment (rickwierenga) [reply Makefile:23]: wdym less strict about line len? why not always <100?

**Comment by rickwierenga** [comment]
we need to check the lint except comments

File: .github/workflows/lint.yml [hunk 3]
```diff
@@ -22,5 +22,7 @@ jobs:
           python-version: 3.9
       - name: Install Dependencies
         run: pip install -e '.[dev]'
-      - name: Run Pytest
+      - name: Run linting check
         run: make lint
+      - name: Check formatting
+        run: make format --check
```
Comment (rickwierenga) [inline .github/workflows/lint.yml:28]:
```suggestion
      - name: Check formatting
        run: make lint
```

Comment (jkhales) [reply .github/workflows/lint.yml:28]: i left it as previous, because people may want to use `make format` to actually apply the changes.

**Comment by rickwierenga** [comment]
some lines are longer than 100 characters

```
./tools/make_resources/tc/make_lc.py:87 has 104 characters
./tools/make_resources/tc/make.py:244 has 130 characters
./tools/make_resources/tc/make.py:257 has 132 characters
./pylabrobot/pumps/agrowpumps/agrowdosepump_tests.py:79 has 115 characters
./pylabrobot/plate_reading/seemstowork.py:99 has 106 characters
./pylabrobot/plate_reading/seemstowork.py:540 has 106 characters
./pylabrobot/plate_reading/fuzz.py:7 has 124 characters
./pylabrobot/plate_reading/fuzz.py:12 has 128 characters
./pylabrobot/plate_reading/fuzz.py:17 has 119 characters
./pylabrobot/liquid_handling/backends/hamilton/STAR.py:1447 has 109 characters
./pylabrobot/liquid_handling/backends/hamilton/STAR.py:1527 has 109 characters
./pylabrobot/resources/ml_star/tip_carriers.py:541 has 107 characters
./pylabrobot/resources/ml_star/tip_carriers.py:761 has 106 characters
./pylabrobot/resources/ml_star/tip_carriers.py:783 has 117 characters
./pylabrobot/resources/ml_star/tip_carriers.py:805 has 105 characters
./pylabrobot/resources/ml_star/tip_carriers.py:827 has 138 characters
./pylabrobot/resources/ml_star/tip_carriers.py:849 has 126 characters
./pylabrobot/resources/tecan/tip_racks.py:247 has 129 characters
./pylabrobot/resources/tecan/tip_racks.py:276 has 134 characters
./pylabrobot/resources/tecan/tip_racks.py:305 has 112 characters
./pylabrobot/resources/tecan/tip_racks.py:650 has 127 characters
./pylabrobot/resources/tecan/tip_racks.py:766 has 133 characters
./pylabrobot/resources/tecan/plates.py:64 has 128 characters
./pylabrobot/resources/tecan/plates.py:770 has 131 characters
```

**Comment by rickwierenga** [comment]
i hope this will save a lot of time going forward

**Pull request merged**